### PR TITLE
Silencing TensorFlow warning

### PR DESCRIPTION
**Pull request by SamFerracin**
**Description of the Change:**
Using python built-in logger, this PR silences the annoying `ComplexWarning`s coming from `tf.cast`.
Additionally, it gets rid of tensorflow's import messages.

### CODE REVIEW

File: mrmustard/math/backend_tensorflow.py [hunk 1]
```diff
@@ -14,14 +14,20 @@
...
 from typing import Callable, List, Optional, Sequence, Tuple, Union
 
+import logging
 import numpy as np
+
+logging.getLogger("tensorflow").setLevel(logging.ERROR)
 import tensorflow as tf
 import tensorflow_probability as tfp
 
+logging.getLogger("tensorflow").setLevel(logging.INFO)
```
Comment (ziofil) [inline mrmustard/math/backend_tensorflow.py:28]: do we need both line 28 and 24?

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -74,8 +74,8 @@ which uses the old Numba code. When setting to a higher value, the new Julia cod
   Hermite polynomials over a batch of B vectors.
   [(#308)](https://github.com/XanaduAI/MrMustard/pull/308)
 
-* Changed the ``cast`` functions in the numpy and tensorflow backends to avoid ``ComplexWarning``s.
-  [(#307)](https://github.com/XanaduAI/MrMustard/pull/307)
+* Added suite to filter undesired warnings, and used it to filter tensorflows' ``ComplexWarning``s.
```
Comment (ziofil) [inline .github/CHANGELOG.md:77]:
```suggestion
* Added suite to filter undesired warnings, and used it to filter tensorflow's ``ComplexWarning``s.
```

**Review by ziofil — APPROVED** [review]
cool!

**Review by sylviemonet — APPROVED** [review]
Works for me! Keep in mind that we have silenced them!!

**Pull request merged**